src/components/modals/SelectSpeakerModal.tsx

```diff
@@ -0,0 +1,97 @@
+import { useState, useLayoutEffect } from 'react';
+import {
+    Text,
+    Modal,
+    ModalOverlay,
+    ModalContent,
+    ModalHeader,
+    ModalCloseButton,
+    ModalBody,
+    ModalFooter,
+    Button,
+    Checkbox,
+    Box
+} from "@chakra-ui/react";
+import { OnlySelectField } from '../sideBarContentEdit/SelectField';
+import { useGetMetadata } from '@/services/api/transcripts/useGetMetadata';
+
+type Props = {
+    selectedSpeakerElement: any;
+    isOpen: boolean;
+    onSetSpeakerName: (newSpeakerName: string | null, isUpdateAllSpeakerInstances: boolean) => void;
+    onClose: () => void;
+};
+
+const SelectSpeakerModal = ({ selectedSpeakerElement, onSetSpeakerName, onClose: closeModal }: Props) => {
+    const oldSpeakerName = selectedSpeakerElement?.speaker || ''
+    const [newSpeakerName, setNewSpeakerName] = useState(oldSpeakerName);
+    const [isUpdateAllSpeakerInstances, setIsUpdateAllSpeakerInstances] = useState(false);
+    const { data: allMetadata } = useGetMetadata();
+
+
+    useLayoutEffect(() => {
+        if (selectedSpeakerElement) {
+            setNewSpeakerName(oldSpeakerName);
+            setIsUpdateAllSpeakerInstances(false);
+        }
+    }, [selectedSpeakerElement]);
+
+    const handleSave = () => {
+        onSetSpeakerName(newSpeakerName, isUpdateAllSpeakerInstances);
+        closeModal();
+    };
+
+    const updateSpeaker = (speakers: string[]) => {
+        setNewSpeakerName(speakers[0])
```

Comment:
This was related with my comment [here](https://github.com/bitcointranscripts/transcription-review-front-end/pull/297#discussion_r1684121325). But I since changed the implementation and the component can work without extra flags for both an array of speakers and single speakers.